prov/gni/src/gnix_msg.c

```diff
@@ -2573,6 +2621,8 @@ ssize_t _gnix_recv(struct gnix_fid_ep *ep, uint64_t buf, size_t len,
 		req->msg.recv_md[0] = md;
 		req->msg.send_iov_cnt = req->msg.recv_iov_cnt = 1;
 		req->msg.recv_flags = flags;
+		if (flags & FI_MULTI_RECV)
+			req->msg.recv_flags |= GNIX_MSG_MULTI_RECV_SUP;
```

Comment:
we discussed this with the general community and it was decided ```FI_MULTI_RECV``` only made sense for single contiguous buffer for receipt of messages in a multi-receive approach.